clang/test/CXX/module/module.interface/p1.cpp

```diff
@@ -35,4 +26,19 @@ export { export int f; } // expected-error {{within another export declaration}}
 
 module :private; // expected-note {{private module fragment begins here}}
 export int priv; // expected-error {{export declaration cannot be used in a private module fragment}}
-#endif
+
+//--- M.cppm
+export module M;
+export int b;
+namespace N {
+  export int c;
+}
+
+//--- impl.cpp
+module M; // #M
+
+export int b2; // expected-error {{export declaration can only be used within a module purview}}
```

Comment:
I know this commit is just moving stuff around, but I wanted to note that this diagnostic is incorrectly worded. This line is in a module purview (https://eel.is/c++draft/module#unit-5).  This should say "within the purview of a module interface" for this case, or just "within a module interface" to be less pedantic.

No need to fix this here though.